lib/resolveurl/plugins/waaw.py

```diff
@@ -22,146 +24,27 @@ class WaawResolver(ResolveUrl):
-                            except:
-                                value_var = ""
-                        params[key] = value_var
-
-                params = urllib.urlencode(params)
-                headers.update({'X-Requested-With': 'XMLHttpRequest', 'Referer': player_url})
-                data = ""
-                data = self.net.http_GET("http://hqq.watch/player/get_md5.php?" + params, headers=headers).content
-                url_data = json.loads(data)
-                media_url = "https:" + self.tb(url_data["obf_link"].replace("#", "")) + ".mp4.m3u8"
-
-                if media_url:
-                    del headers['X-Requested-With']
-                    headers.update({'Origin': 'https://hqq.watch'})
-                    return media_url + helpers.append_headers(headers)
-
-            except Exception as e:
-                raise ResolverError(e)
-
-        raise ResolverError('Video not found')
-
-    def tb(self, b_m3u8_2):
-        j = 0
-        s2 = ""
-        while j < len(b_m3u8_2):
-            s2 += "\\u0" + b_m3u8_2[j:(j + 3)]
-            j += 3
-
-        return s2.decode('unicode-escape').encode('ASCII', 'ignore')
-
-    ## loop2unobfuscated
-    def jswise(self, wise):
-        while True:
-            wise = re.search("var\s.+?\('([^']+)','([^']+)','([^']+)','([^']+)'\)", wise, re.DOTALL)
-            if not wise: break
-            ret = wise = self.js_wise(wise.groups())
-
-        return ret
-
-    ## js2python
-    def js_wise(self, wise):
-        w, i, s, e = wise
-
-        v0 = 0;
-        v1 = 0;
-        v2 = 0
-        v3 = [];
-        v4 = []
-
-        while True:
-            if v0 < 5:
-                v4.append(w[v0])
-            elif v0 < len(w):
-                v3.append(w[v0])
-            v0 += 1
-            if v1 < 5:
-                v4.append(i[v1])
-            elif v1 < len(i):
-                v3.append(i[v1])
-            v1 += 1
-            if v2 < 5:
-                v4.append(s[v2])
-            elif v2 < len(s):
-                v3.append(s[v2])
-            v2 += 1
-            if len(w) + len(i) + len(s) + len(e) == len(v3) + len(v4) + len(e): break
-
-        v5 = "".join(v3);
-        v6 = "".join(v4)
-        v1 = 0
-        v7 = []
-
-        for v0 in range(0, len(v3), 2):
-            v8 = -1
-            if ord(v6[v1]) % 2: v8 = 1
-            v7.append(chr(int(v5[v0:v0 + 2], 36) - v8))
-            v1 += 1
-            if v1 >= len(v4): v1 = 0
-
-        return "".join(v7)
-
+                   'x-requested-with': 'XMLHttpRequest'}
+        params = (
+        	('ver', '0'),
+        	('secure', '0'),
+        	('adb', '0/'),
+        	('v', media_id),
+        	('token', ''),
+        	('gt', ''),)	   
+        params = urllib.urlencode(params)
+        rurl = "https://hqq.tv/player/get_md5.php?" + params  
+        str_url = self.net.http_HEAD(rurl, headers=headers).get_url()
+        return str_url+'.mp4.m3u8' + helpers.append_headers(headers)
```

Comment:
needs `headers.pop('x-requested-with)` before this line. Please check

Reply:
It's not necessary.

Reply:
It is good practice to not send unnecessary headers when requesting stream url. "X-Requested-With" generally is never used with mp4 or m3u8 urls

Reply:
please include `headers.pop('x-requested-with')` before the return statement